server/sockets/helpers/index.js

```diff
@@ -88,11 +69,15 @@ const updateListOnDashboardAndCohortView = (
             const currentList = responseWithList(doc, id);
 
             if (cohortViewClients.has(viewerId)) {
-              socket.broadcast
-                .to(cohortViewClients.get(viewerId))
-                .emit(ListActionTypes.FETCH_META_DATA_SUCCESS, {
-                  [listId]: { ...currentList }
-                });
+              const { viewId, socketId } = cohortViewClients.get(viewerId);
+
+              if (viewId === cohortId.toString()) {
+                socket.broadcast
+                  .to(socketId)
+                  .emit(ListActionTypes.FETCH_META_DATA_SUCCESS, {
+                    [listId]: { ...currentList }
+                  });
+              }
```

Comment:
the same here,check previous comment

Reply:
we have to first check if exist then destructure to certain props as we discussed
but I simplify code and removed checking the size of clients maps